core/test/variable_trigonometry_test.cpp

```diff
@@ -8,323 +8,226 @@
 
 using namespace scipp;
 using namespace scipp::core;
+using namespace scipp::units;
 
-TEST(VariableTrigonometryTest, sin) {
-  const auto rad = makeVariable<double>(
-      Dims(), Shape(), units::Unit(units::rad), Values{pi<double>});
-  const auto deg = makeVariable<double>(Dims(), Shape(),
-                                        units::Unit(units::deg), Values{180.0});
-
-  const auto expected =
-      makeVariable<double>(Dims(), Shape(), units::Unit(units::dimensionless),
-                           Values{sin(pi<double>)});
-
-  EXPECT_EQ(sin(rad), expected);
-  EXPECT_EQ(sin(deg), expected);
+TEST(VariableTrigonometryTest, sin_rad) {
+  const auto var = makeVariable<double>(Values{pi<double>}, Unit{units::rad});
+  EXPECT_EQ(sin(var), makeVariable<double>(Values{std::sin(pi<double>)}));
 }
 
-TEST(VariableTrigonometryTest, sin_in_place_full) {
-  auto rad =
-      makeVariable<double>(Dims{Dim::X}, Shape{3}, units::Unit(units::rad),
-                           Values{0.0, pi<double>, 2.0 * pi<double>});
-  auto deg =
-      makeVariable<double>(Dims{Dim::X}, Shape{3}, units::Unit(units::deg),
-                           Values{0.0, 180.0, 360.0});
-
-  auto rad_view = sin(rad, rad);
-  auto deg_view = sin(deg, deg);
-
-  const auto expected = makeVariable<double>(
-      Dims{Dim::X}, Shape{3},
-      Values{sin(0.0), sin(pi<double>), sin(2.0 * pi<double>)});
-
-  EXPECT_EQ(rad, expected);
-  EXPECT_EQ(rad_view, rad);
-  EXPECT_EQ(rad_view.underlying(), rad);
-
-  EXPECT_EQ(deg, expected);
-  EXPECT_EQ(deg_view, deg);
-  EXPECT_EQ(deg_view.underlying(), deg);
+TEST(VariableTrigonometryTest, sin_deg) {
```

Comment:
Do you consider it necessary to have this test in addition to `sin_rad` above? The tests of the kernel should ensure that this works, so maybe the retesting could be avoided?

Reply:
Yes, as the kernel only copes with input in radians, so the logic to deal with either unit is in `scipp::sin()` itself. As such this is the lowest point at which correctly handling either degrees or radians could be tested.

Reply:
Good point, I missed that. 👍